datafusion/functions/src/regex/regexpreplace.rs

```diff
@@ -313,25 +322,24 @@ macro_rules! fetch_string_arg {
 fn _regexp_replace_static_pattern_replace<T: OffsetSizeTrait>(
     args: &[ArrayRef],
 ) -> Result<ArrayRef> {
-    let string_array = as_generic_string_array::<T>(&args[0])?;
-    let array_size = string_array.len();
+    let array_size = args[0].len();
     let pattern = fetch_string_arg!(
         &args[1],
         "pattern",
-        T,
+        i32,
```

Comment:
I don't understand why thiw was changed from `T` to `i32` 🤔

Reply:
Because the second input is always must be Utf8 (as shown in the Signature above).
I'm trying to make `T` to only be used on the `arg[0]`, so that later we can test different types of `arg[0]`, e.g., `Utf8View`, `LargeUtf8`